### ISSUE TITLE
Ability to explicitly suppress diffs (mark paths as binary)

### ISSUE THREAD

**Issue by zbjornson** [github/linguist#4262]
git allows you to mark files as binary by adding `<path> -diff` to .gitattributes: https://git-scm.com/docs/gitattributes#_marking_files_as_binary. Linguist doesn't appear to obey that.

## Preliminary Steps

Please confirm you have...
- [x] reviewed [How Linguist Works](https://github.com/github/linguist#how-linguist-works),
- [x] reviewed the [Troubleshooting](https://github.com/github/linguist#troubleshooting) docs,
- [x] considered implementing an [override](https://github.com/github/linguist#overrides),
- [x] verified an issue has not already been logged for your issue ([linguist issues](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+repo%3Agithub/linguist)).

I see a few related issues here but none explicitly mention obeying the `-diff` setting or they have different intents.

## Problem Description

We have some test fixtures that are a mix of binary and text content. (Since they are test fixtures, so we don't want to abuse the "vendored," "documentation" or other linguist overrides that might suppress the diff.) Linguist decides that some of these files are text and show them in the diff stats (i.e. the `+xxx -yyy` bit in a PR or commit). `git show` correctly says "Binary files xxx and yyy differ".

### URL of the affected repository:
Private. I can make a public repro if necessary.

### Last modified on:
2018/09/01

### Expected language:
(binary)

### Detected language:
(something not binary)

**Comment by zbjornson** [comment]
Oops, sorry. The goal is for PRs to say "binary file" (suppress the diff) and not count the lines in those files in PR diff stats ("+xxx -yyy" at the top of the PR page).

**Comment by pchaigno** [comment]
@lildude Should we recognize the `diff` attribute in Linguist? Looks like we could simply add support for it in [`lazy_blob.rb`](https://github.com/github/linguist/blob/a5df9a00ab7c9828bd7038bb9f9bd5e56d325dc9/lib/linguist/lazy_blob.rb#L8) and maybe move [`binary?`](https://github.com/github/linguist/blob/a5df9a00ab7c9828bd7038bb9f9bd5e56d325dc9/lib/linguist/blob_helper.rb#L130) from `blob_helper.rb` to `lazy_blob.rb`.

**Comment by lildude** [comment]
> @lildude Should we recognize the `diff` attribute in Linguist? Looks like we could simply add support for it in lazy_blob.rb and maybe move binary? from `blob_helper.rb` to `lazy_blob.rb`.

I've discussed this with the team that deals with diffs on GitHub.com and the suppression in the diffs can't be implemented in Linguist as the code involved doesn't use `Linguist::BlobHelper#binary?` when rendering diffs. Rather they use the information they get from the diff output.

As such, I've opened an internal issue for the `-diff` attribute to be considered when rendering diffs and this will be dealt with by the team in due course.

As there is nothing we can do in Linguist, I'm going to close this issue.

Thanks for raising this to our attention @zbjornson 🙇

**Comment by ted-mccarthyfinch** [comment]
I know this is closed, but have there been any updates on this?

**Comment by TedChenNZ** [comment]
In case anybody else wants to look at how to suppress diffs for github, take a look at this issue https://github.com/github/linguist/issues/2206

TLDR you can suppress the diff on both github and git by doing something like 

`folder_to_suppress_diffs/**/* binary linguist-generated`
